src/backend/app.js

```diff
@@ -21,6 +23,51 @@ app.use(cors());
 
 router.use("/meals", mealsRouter);
 
+
+app.get("/:type", async (req, res) => {
+  const { type } = req.params;
+  const now = new Date();
+
+  let query;
+  
+  // Routes and their responses
+  switch (type) {
+    case "future-meals":
+      query = knex.select('*').from('meal').where('when_date', '>', now);
+      break;
+    case "past-meals":
+      query = knex.select('*').from('meal').where('when_date', '<', now);
+      break;
+    case "all-meals":
+      query = knex.select('*').from('meal').orderBy('id');
+      break;
+    case "first-meal":
+      query = knex.select('*').from('meal').where("id", 1).first();
```

Comment:
I have a few questions regarding this query:
1) Are you sure you have id=1? What's going to happen if you delete a meal with id=1 before executing this query? 
2) How many meals you're expecting to have with id=1 if you've added first() in the end of the query? Id should be unique, that's why when you specify it in query you always should get a single item.